src/movies.js

```diff
@@ -1,16 +1,62 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(movies) {
+    const directors = movies.map(function (movie) {
+        return movie.director
+    });
+    return directors
+}
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(movies) {
+    const dramaSpielberg = movies.filter(function (movie) {
+        if (movie.director === 'Steven Spielberg' && movie.genre.includes('Drama'))
+            return movie.director;
+    });
+    return dramaSpielberg.length;
+}
+
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage(movies) {
+    const average = movies.reduce(function (acc, val) {
+        if (val.rate !== "" && ('rate' in val)) {
```

Comment:
Nice!
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in